src/Microsoft.Extensions.Logging.Abstractions/LoggerMessage.cs

```diff
@@ -0,0 +1,343 @@
+        /// <param name="logLevel">The <see cref="LogLevel"/></param>
+        /// <param name="eventId">The event id</param>
+        /// <param name="eventName">The event name</param>
+        /// <param name="formatString">The named format string</param>
+        public static void Define<T1, T2>(out Action<ILogger, T1, T2, Exception> message, LogLevel logLevel, int eventId, string eventName, string formatString)
+        {
+            var formatter = new LogValuesFormatter("{EventName}: " + formatString);
+            Func<object, Exception, string> callback = (state, error) => formatter.Format(((LogValues<string, T1, T2>)state).ToArray());
+
+            message = (logger, arg1, arg2, exception) =>
+            {
+                if (logger.IsEnabled(logLevel))
+                {
+                    logger.Log(logLevel, eventId, new LogValues<string, T1, T2>(formatter, eventName, arg1, arg2), exception, LogValues<string, T1, T2>.Callback);
+                }
+            };
+        }
+
+        /// <summary>
+        /// Creates a delegate which can be used for logging a message.
+        /// </summary>
+        /// <typeparam name="T1">The parameter passed to the named format string</typeparam>
+        /// <typeparam name="T2">The parameter passed to the named format string</typeparam>
+        /// <typeparam name="T3">The parameter passed to the named format string</typeparam>
+        /// <param name="message">A delegate which creates a log message</param>
+        /// <param name="logLevel">The <see cref="LogLevel"/></param>
+        /// <param name="eventId">The event id</param>
+        /// <param name="eventName">The event name</param>
+        /// <param name="formatString">The named format string</param>
+        public static void Define<T1, T2, T3>(out Action<ILogger, T1, T2, T3, Exception> message, LogLevel logLevel, int eventId, string eventName, string formatString)
+        {
+            var formatter = new LogValuesFormatter("{EventName}: " + formatString);
+            Func<object, Exception, string> callback = (state, error) => formatter.Format(((LogValues<string, T1, T2, T3>)state).ToArray());
+
+            message = (logger, arg1, arg2, arg3, exception) =>
+            {
+                if (logger.IsEnabled(logLevel))
+                {
+                    logger.Log(logLevel, eventId, new LogValues<string, T1, T2, T3>(formatter, eventName, arg1, arg2, arg3), exception, LogValues<string, T1, T2, T3>.Callback);
+                }
+            };
+        }
+
+        private class LogValues : ILogValues
+        {
+            public static Func<object, Exception, string> Callback = (state, exception) => ((LogValues)state)._formatter.Format(((LogValues)state).ToArray());
+
+            private static IEnumerable<KeyValuePair<string, object>> _getValues = new KeyValuePair<string, object>[0];
+            private static object[] _toArray = new object[0];
```

Comment:
:+1: looks like `OriginalFormat` should be included based on how `LogValuesFormatter` implements this. Updated the test accordingly.